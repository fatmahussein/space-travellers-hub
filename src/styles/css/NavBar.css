.navbar {
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1rem;
  border-bottom: 1px solid gray;
}

.navbar .logoInfo {
  display: flex;
  align-items: center;
  gap: 1rem;
}

h1 {
  font-size: 30px;
}

.navbar img {
  width: 40px;
}

.links {
  display: flex;
  gap: 1rem;
  font-size: 20px;
}

.links a {
  color: blue;
}

.active-link {
  text-decoration: dotted;
}
.border{
  border: 1px solid black;
}