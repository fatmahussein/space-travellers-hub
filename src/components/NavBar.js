import React from 'react';
import { NavLink } from 'react-router-dom';
import planetImg from '../Assets/images/planet.png';
import '../styles/css/NavBar.css';

const NavBar = () => {
  const handleActive = (nav) => (nav.isActive ? '' : 'active-link');
  const result = (
    <header className="navbar">
      <div className="logoInfo">
        <img src={planetImg} alt="" />
        <h1>Space Traveller&apos;s Hub</h1>
      </div>
      <div className="links">
        <NavLink className={(nav) => handleActive(nav)} to="/">Rockets</NavLink>
        <NavLink className={(nav) => handleActive(nav)} to="/missions">Missions</NavLink>
        <NavLink className={(nav) => handleActive(nav)} to="/dragons">Dragons</NavLink>
        <div className="border" />
        <NavLink className={(nav) => handleActive(nav)} to="/Myprofile">My Profile</NavLink>
      </div>
    </header>
  );

  return result;
};

export default NavBar;
