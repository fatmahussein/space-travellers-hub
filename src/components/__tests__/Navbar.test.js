import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import NavBar from '../NavBar';

const mockStore = configureStore([]);
describe('Navbar', () => {
  test('Navbar rendered correctly', () => {
    const store = mockStore({
      rockets: {
        rocket: [
          {
            name: 'Test',
            description: 'Test Description',
            reserved: false,
            id: 'test-id',
          },
        ],
        isLoading: false,
        isError: false,
      },
    });
    const navbar = render(
      <Provider store={store}>
        <Router>
          <NavBar />
        </Router>
      </Provider>,
    );
    expect(navbar).toMatchSnapshot();
  });
});
