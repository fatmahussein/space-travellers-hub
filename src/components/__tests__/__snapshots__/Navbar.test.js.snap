// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Navbar rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="navbar"
      >
        <div
          class="logoInfo"
        >
          <img
            alt=""
            src="planet.png"
          />
          <h1>
            Space Traveller's Hub
          </h1>
        </div>
        <div
          class="links"
        >
          <a
            aria-current="page"
            class=""
            href="/"
          >
            Rockets
          </a>
          <a
            class="active-link"
            href="/missions"
          >
            Missions
          </a>
          <a
            class="active-link"
            href="/dragons"
          >
            Dragons
          </a>
          <div
            class="border"
          />
          <a
            class="active-link"
            href="/Myprofile"
          >
            My Profile
          </a>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="navbar"
    >
      <div
        class="logoInfo"
      >
        <img
          alt=""
          src="planet.png"
        />
        <h1>
          Space Traveller's Hub
        </h1>
      </div>
      <div
        class="links"
      >
        <a
          aria-current="page"
          class=""
          href="/"
        >
          Rockets
        </a>
        <a
          class="active-link"
          href="/missions"
        >
          Missions
        </a>
        <a
          class="active-link"
          href="/dragons"
        >
          Dragons
        </a>
        <div
          class="border"
        />
        <a
          class="active-link"
          href="/Myprofile"
        >
          My Profile
        </a>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
